include/ddynamic_reconfigure/ddynamic_reconfigure.h

```diff
@@ -131,18 +131,34 @@ class DDynamicReconfigure
   typedef boost::function<void()> UserCallbackType;
 
   /**
-   * @brief setUserCallback An optional callback that will be called whenever a value is
+   * @brief setPreUpdateCallback An optional callback that will be called right before any value is
    * changed
    */
-  virtual void setUserCallback(const UserCallbackType &callback);
+  virtual void setPreUpdateCallback(const UserCallbackType &callback);
 
-  virtual void clearUserCallback();
+  virtual void clearPreUpdateCallback();
 
+  /**
+   * @brief setPostUpdateCallback An optional callback that will be called right after any value is
+   * changed.
+   */
+  virtual void setPostUpdateCallback(const UserCallbackType &callback);
+
+  virtual void clearPostUpdateCallback();
 
   /**
    * Deprecated functions. For backwards compatibility, cannot be a template for legacy
    * reasons
    */
+
+  /**
+   * @brief setUserCallback An optional callback that will be called whenever a value is
+   * changed. This is equivalent to the setPostUpdateCallback.
+   */
+  virtual void setUserCallback(const UserCallbackType &callback);
+
+  virtual void clearUserCallback();
```

Comment:
Just a comment. Are you keeping them for backward compatibility?

Reply:
Yes, this change is fully backward compatible.

Reply:
I see 👍